patch.js

```diff
@@ -0,0 +1,755 @@
+        if (res) this._emitReady();
+        next(err, res);
+      }.bind(this));
+    }.bind(this), function(error, results) {
+      results.forEach(function(item) {
+        if (!item || this.closed) return;
+        this.add(sysPath.dirname(item), sysPath.basename(_origAdd || item));
+      }, this);
+    }.bind(this));
+  }
+
+  return this;
+};
+
+// Public method: Close watchers or start ignoring events from specified paths.
+
+// * paths     - string or array of strings, file/directory paths and/or globs
+
+// Returns instance of FSWatcher for chaining.
+FSWatcher.prototype.unwatch = function(paths) {
+  if (this.closed) return this;
+  paths = flatten(arrify(paths));
+
+  paths.forEach(function(path) {
+    // convert to absolute path unless relative path already matches
+    if (!isAbsolute(path) && !this._closers[path]) {
+      if (this.options.cwd) path = sysPath.join(this.options.cwd, path);
+      path = sysPath.resolve(path);
+    }
+
+    this._closePath(path);
+
+    this._ignoredPaths[path] = true;
+    if (path in this._watched) {
+      this._ignoredPaths[path + '/**'] = true;
+    }
+
+    // reset the cached userIgnored anymatch fn
+    // to make ignoredPaths changes effective
+    this._userIgnored = null;
+  }, this);
+
+  return this;
+};
+
+// Public method: Close watchers and remove all listeners from watched paths.
+
+// Returns instance of FSWatcher for chaining.
+FSWatcher.prototype.close = function() {
+  if (this.closed) return this;
+
+  this.closed = true;
+  Object.keys(this._closers).forEach(function(watchPath) {
+    this._closers[watchPath].forEach(function(closer) {
+      closer();
+    });
+    delete this._closers[watchPath];
+  }, this);
+  this._watched = Object.create(null);
+
+  this.removeAllListeners();
+  return this;
+};
+
+// Public method: Expose list of watched paths
+
+// Returns object w/ dir paths as keys and arrays of contained paths as values.
+FSWatcher.prototype.getWatched = function() {
+  var watchList = {};
+  Object.keys(this._watched).forEach(function(dir) {
+    var key = this.options.cwd ? sysPath.relative(this.options.cwd, dir) : dir;
+    watchList[key || '.'] = Object.keys(this._watched[dir]._items).sort();
+  }.bind(this));
+  return watchList;
+};
+
+// Attach watch handler prototype methods
+function importHandler(handler) {
+  Object.keys(handler.prototype).forEach(function(method) {
+    FSWatcher.prototype[method] = handler.prototype[method];
+  });
+}
+importHandler(NodeFsHandler);
+if (false) importHandler(FsEventsHandler);
+
+// Export FSWatcher class
+exports.FSWatcher = FSWatcher;
+
+// Public function: Instantiates watcher with paths to be tracked.
+
+// * paths     - string or array of strings, file/directory paths and/or globs
+// * options   - object, chokidar options
+
+// Returns an instance of FSWatcher for chaining.
+exports.watch = function(paths, options) {
+  return new FSWatcher(options).add(paths);
+};
+
+`
+);
```

Comment:
@hiroppy yes, problem(s) in `fsevents`, I patched `chokidar` does not use `fsevents` and tests passed.

Reply:
Now macos users would probably see webpack consuming 1GB of memory and tons of CPU.

Reply:
@paulmillr Firstly, this is not true, I can take measurements again and demonstrate them to you.  Secondly, I don’t see a desire to correct a bug or help us find it, so I don’t see any other choice how to turn off `fsevents`. I am always open and aimed at improvements and fix bugs, but for some reason you react negatively to my attempts to find a compromise.

Reply:
That's because we've already fixed the bug in chokidar 3 — and you should update. There is no "unwilingness", there is just desire to not do the same work twice.

Go ahead and remove fsevents. You'll notice some bug reports in a couple weeks.

Reply:
@paulmillr Why are you so negative to me, have I really done something bad to you? All I'm trying to do is find the error and fix it.